src/NzbDrone.Core/Indexers/TorrentRssIndexer/TorrentRssSettingsDetector.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Xml;
+using System.Xml.Linq;
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Indexers.Exceptions;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.Indexers.TorrentRssIndexer
+{
+    public class TorrentRssSettingsDetector
+    {
+        protected readonly Logger _logger = LogManager.GetCurrentClassLogger();
+
+        protected TorrentRssIndexerSettings _settings;
+
+        protected Func<IndexerRequest, IndexerResponse> _fetchIndexerResponseFunc;
+
+        /// <summary>
+        /// Detect settings for Parser, based on URL
+        /// </summary>
+        /// <param name="settings">Indexer Settings to use for Parser</param>
+        /// <param name="fetchIndexerResponseFunc">Func to retrieve Feed</param>
+        /// <returns>Parsed Settings or <c>null</c></returns>
+        public TorrentRssIndexerParserSettings Detect(TorrentRssIndexerSettings settings, Func<IndexerRequest, IndexerResponse> fetchIndexerResponseFunc)
```

Comment:
Can we just hit http client directly? We're using the same implementation here so a func seems like overkill. Alternatively we could pass the result to this function, but that adds another level to go through.